grpcbp/server_middlewares_test.go

```diff
@@ -0,0 +1,193 @@
+package grpcbp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	pb "github.com/grpc-ecosystem/go-grpc-middleware/testing/testproto"
+	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/mqsend"
+	"github.com/reddit/baseplate.go/tracing"
+	"github.com/reddit/baseplate.go/transport"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	testTimeout = time.Millisecond * 100
+)
+
+func TestInjectServerSpanInterceptor(t *testing.T) {
+	// create test server with InjectServerSpanInterceptor
+	l, _ := initServer(t, grpc.UnaryInterceptor(InjectServerSpanInterceptor()))
+
+	// create test client
+	conn := initClient(t, l)
+
+	// instantiate gRPC client
+	client := pb.NewTestServiceClient(conn)
+
+	// set up recorder to validate span recording
+	mmq := initTracing(t)
+
+	t.Run("span-success", func(t *testing.T) {
+		// send request to server
+		_, err := client.Ping(context.Background(), &pb.PingRequest{})
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
same here and below, use `if err := ...` when you don't need any of the values past the if chain